src/pages/index.js

```diff
@@ -33,155 +36,168 @@ const initialCards = [
+updateAvatarValidator.enableValidation();
+updateAvatarValidator.disableButton();
 
 /* const closeModal = (modalWindow) => {
   modalWindow.classList.remove("modal_open");
   document.removeEventListener("keydown", handleEscapeKey);
 }; */
 
 function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleCardClick);
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleCardClick,
+    deleteCardSubmit,
+    handleLikeClick
+  );
   return cardElement.getView();
 }
 
-/* con = (data, wrap) => {
-  const card = createCard(data);
-  wrap.prepend(card);
-}; */
-
 const profileName = document.querySelector(".profile__name");
 const profileDescription = document.querySelector(".profile__description");
-
+const avatar = document.querySelector(".profile__image");
 const profileNameInput = document.querySelector("#profile-name-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-const userData = new UserInfo(profileName, profileDescription);
+function handleProfileEditSubmit(userProfile) {
+  popupWithEditForm.renderLoading(true);
+
+  api
+    .updateUserInfo(userProfile.name, userProfile.link)
+    .then(() => {
+      userData.setUserInfo(userProfile.name, userProfile.link);
+    })
+    .then(() => {
+      editFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      popupWithEditForm.renderLoading(false);
+    });
 
-function handleProfileEditSubmit() {
-  userData.setUserInfo(profileNameInput.value, profileDescriptionInput.value);
   popupWithEditForm.close();
 }
 
+function handleAvatarSubmit(data) {
+  avatarCardPopUp.renderLoading(true);
+
+  api
+    .updateUserAvatar(avatarUrlInput.value)
+    .then((res) => {
+      userData.setUserAvatar(res.avatar);
+    })
+    .then(() => {
+      console.log("Avatar has been updated");
+      updateAvatarValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarCardPopUp.renderLoading(false);
+    });
+
+  avatarCardPopUp.close();
+}
+
 profileEditButton.addEventListener("click", () => {
   popupWithEditForm.open();
   const userProfile = userData.getUserInfo();
   profileNameInput.value = userProfile.name;
   profileDescriptionInput.value = userProfile.job;
+  
 });
 
 const name = cardTitleInput.value;
 const link = cardUrlInput.value;
 
-
 function handleAddCardSubmit(data) {
-  const card = createCard(data);
-  cardSection.addItem(card);
+  popupWithAddForm.renderLoading(true);
+  api
+    .createCard(data.name, data.link)
+    .then((results) => {
+      const card = createCard(results);
+      cardSection.addItem(card);
+    })
+    .catch((err) => {
+      console.error(`The error is ${err}`);
+    })
+    .finally(() => {
+      popupWithAddForm.renderLoading(false);
+    });
   popupWithAddForm.close();
+  addFormValidator.disableButton();
```

Comment:
These 2 lines should be in `then`


You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


And disabling as well (after closing)